components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -496,6 +497,55 @@ class GeckoEngine(
         })
     }
 
+    /**
+     * See [Engine.isProfilerActive].
+     */
+    override fun isProfilerActive(): Boolean {
+        return runtime.profilerController.isProfilerActive
+    }
+
+    /**
+     * See [Engine.getProfilerTime].
+     */
+    override fun getProfilerTime(): Double? {
+        return runtime.profilerController.profilerTime
+    }
+
+    /**
+     * See [Engine.addProfilerMarker].
+     */
+    override fun addProfilerMarker(markerName: String, startTime: Double?, endTime: Double?, text: String?) {
+        runtime.profilerController.addMarker(markerName, startTime, endTime, text)
+    }
+
+    /**
+     * See [Engine.addProfilerMarker].
+     */
+    override fun addProfilerMarker(markerName: String, startTime: Double?, text: String?) {
+        runtime.profilerController.addMarker(markerName, startTime, text)
+    }
+
+    /**
+     * See [Engine.addProfilerMarker].
+     */
+    override fun addProfilerMarker(markerName: String, startTime: Double) {
+        runtime.profilerController.addMarker(markerName, startTime)
+    }
+
+    /**
+     * See [Engine.addProfilerMarker].
+     */
+    override fun addProfilerMarker(markerName: String, text: String?) {
+        runtime.profilerController.addMarker(markerName, text)
+    }
+
+    /**
+     * See [Engine.addProfilerMarker].
+     */
+    override fun addProfilerMarker(markerName: String) {
+        runtime.profilerController.addMarker(markerName)
+    }
+
```

Comment:
Nit: I wonder if we should layer this too, like the `GeckoRuntime`, so that the interface of `Engine` does not get too crowded (and it also avoids adding "profiler" to all methods).

Maybe something like a `profiler` property on `Engine`? If it is nullable then you already can use it safely and you do not need to care about no-op/throwing:

```Kotlin
interface Engine {
   val profiler: Profiler?
}


engine.profiler?.addMarker(..)
```

Reply:
Thanks for the recommendation @pocmo, that definitely makes the code better! Pushed the latest code with this change applied. Could you take a look at it again please?